app/src/main/java/com/seay/game_of_thrones/inject/ApplicationModule.java

```diff
@@ -37,4 +42,37 @@ Welcome provideWelcome() {
     ApiService provideApiService(Context context) {
         return new ApiService(context);
     }
+
+    @Provides
+    @Singleton
+    MainDAO provideAccessMainDAO() {
+        return new MainDAO();
+    }
+
+    @Provides
+    @Singleton
+    CharacterRepository provideCharacterRepository() {
+        return new CharacterRepository();
+    }
+
+    @Provides
+    @Singleton
+    JobManager jobManager() {
+        int maxCount = 5;
+        int loadFactor = 3;
+        int minCount = 1;
+
+        Configuration configuration = new Configuration.Builder(application)
+                .consumerKeepAlive(45)
+                .maxConsumerCount(maxCount)
+                .loadFactor(loadFactor)
+                .minConsumerCount(minCount)
+                .injector(job -> {
+                    if (job instanceof GetCharacterListJob) {
```

Comment:
Doing it this way would be super annoying whenever you added a new job. Instead we can use `BaseJob`